builtin/mainmenu/content/contentdb.lua

```diff
@@ -181,14 +181,24 @@ function contentdb.get_package_by_id(id)
 end
 
 
-local function get_raw_dependencies(package)
-	if package.type ~= "mod" then
-		return {}
-	end
-	if package.raw_deps then
-		return package.raw_deps
+local function make_callback_coroutine(fn, callback)
```

Comment:
Might be interesting to add a coroutine http api

Reply:
Implemented in 066e07cb51cf7a443014ae4f5b5119b31289ac18 to give the idea a try.

Differences:

- `core.parse_json` runs on the main thread again. Previously, this PR moved it into `core.handle_async`.

- The difference between the master branch and this PR is smaller than before.

- The overall amount of code is still exactly the same.

- Is this approach more ergonomic? I don't know.

What do you think?

Reply:
Maybe I just don't understand this resumer-pattern, but would `await` ([like here](https://github.com/minetest/minetest/blob/6303334cc2f39732697a1278532cbe66ed481da8/games/devtest/mods/unittests/init.lua#L27)) be a suitable replacement?
Just a suggestion ofc.

Reply:
When using your await function, the problem I have is that the final callback (the one passed into `contentdb.has_hard_deps` / `contentdb.resolve_dependencies` by the API user) is not called. To make this work, the value returned by `coroutine.resume` here

https://github.com/minetest/minetest/blob/6303334cc2f39732697a1278532cbe66ed481da8/games/devtest/mods/unittests/init.lua#L38

would somehow have to get back to the final callback, which brings me back to my "resumer" pattern.

This is my first time working with coroutines. There might of course be a better approach that I haven't seen.